### PR TITLE
Sm/sf command

### PR DESCRIPTION
**Pull request by mshanemc**
> requires other PRs: https://github.com/salesforcecli/plugin-org/pull/543, https://github.com/salesforcecli/plugin-org/pull/549, https://github.com/salesforcecli/plugin-org/pull/550

> when you merge this and put it in `sf` also merge this counter-PR https://github.com/salesforcecli/plugin-env/pull/485

### What does this PR do?

just the sfCommand and style/linter etc.  The renaming and deprecation stuff will be a separate PR
the last PR in this chain moves the sandboxNUTs to a UI-run workflow in GHA

### What issues does this PR fix or reference?
[@W-12083557@](https://gus.lightning.force.com/a07EE00001DTPYDYA5)

### Release Notes

`force:org:create` can no longer be run as `force:org:beta:create`

The following 4 commands are deprecated.
* `force:org:create`:  Use `org:create:scratch`  or `org:create:sandbox`
* `force:org:delete`:  Use `org:delete:scratch`  or `org:delete:sandbox`
* `force:org:status`:  Use `org:resume:sandbox`  
* `force:org:clone`:  Use `org:create:sandbox`

`force:org:open`, `force:org:list`, and `force:org:display` should be run without the word `force`.  

Flag changes
* for all commands
  * The `loglevel` flag is still available but will have no effect of the commands in this plugin
  * `targetusername` flag is now `target-org`
  * `targetdevhubusername` is now `target-dev-hub`
  * `apiversion` is now `api-version`
  * `noprompt` is now `no-prompt`
* org:list
  * `skipconnectionstatus` is now `skip-connection-status`
*  org:open
  * `urlonly` is now `url-only`

### CODE REVIEW

File: src/shared/orgListUtil.ts [hunk 1]
```diff
@@ -256,38 +253,38 @@ export class OrgListUtil {
...
+    });
+
+    const warnings = results.filter((result) => typeof result === 'string') as string[];
+    if (warnings.length) {
+      const logger = await OrgListUtil.retrieveLogger();
+      warnings.forEach((warning) => logger.warn(warning));
     }
 
-    return orgs;
+    return results.filter((result) => typeof result !== 'string') as FullyPopulatedScratchOrgFields[];
```
Comment (peternhale) [inline src/shared/orgListUtil.ts:287]:
``` suggestion
    const logger = await OrgListUtil.retrieveLogger();
    const contentMap = new Map(updatedContents.map((org) => [org.SignupUsername, org]));

    return orgs.filter((scratchOrgInfo) => {
      if (!contentMap.has(scratchOrgInfo.username)) {
        logger.warn(`Can't find ${scratchOrgInfo.username} in the updated contents`);
        return false;
      }
      return true;
    })
      .map((scratchOrgInfo): FullyPopulatedScratchOrgFields => {
        const updatedOrgInfo = contentMap.get(scratchOrgInfo.username)!;
        return {
          ...scratchOrgInfo,
          signupUsername: updatedOrgInfo.SignupUsername,
          createdBy: updatedOrgInfo.CreatedBy.Username,
          createdDate: updatedOrgInfo.CreatedDate,
          devHubOrgId: updatedOrgInfo.devHubOrgId,
          devHubId: updatedOrgInfo.devHubOrgId,
          attributes: updatedOrgInfo.attributes,
          orgName: updatedOrgInfo.OrgName,
          edition: updatedOrgInfo.Edition,
          status: updatedOrgInfo.Status,
          expirationDate: updatedOrgInfo.ExpirationDate,
          isExpired: updatedOrgInfo.Status === 'Deleted',
          namespace: updatedOrgInfo.Namespace
        };
      });
```

File: src/shared/orgListUtil.ts [hunk 2]
```diff
@@ -341,4 +338,36 @@ export class OrgListUtil {
...
+export const identifyActiveOrgByStatus = (org: FullyPopulatedScratchOrgFields): boolean =>
+  'status' in org && org.status === 'Active';
+
+/** Identify the default orgs */
+const identifyDefaultOrgs = (
+  orgInfo: AuthFieldsFromFS,
+  config: JsonMap
+): ExtendedAuthFields | ExtendedAuthFieldsScratch => ({
+  ...orgInfo,
+  ...((config['target-org'] && (orgInfo.username === config['target-org'] || orgInfo.alias === config['target-org'])) ||
```
Comment (peternhale) [inline src/shared/orgListUtil.ts:350]: @mshanemc can/will this be simplified by asking the config aggregator for default target-org and dev-hub?

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/commands/force/org/list.ts [hunk 3]
```diff
@@ -118,95 +109,133 @@ export class OrgListCommand extends SfdxCommand {
...
-      alias: {
-        header: 'ALIAS',
-        get: (data: ExtendedAuthFields): string => data.alias ?? '',
-      },
-      username: { header: 'USERNAME' },
-      orgId: { header: 'ORG ID' },
-    };
+// sort by alias then username
+const comparator = <T extends ExtendedAuthFields | FullyPopulatedScratchOrgFields>(a: T, b: T): number => {
+  const aAlias = (a.alias ?? '').toUpperCase();
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/commands/force/org/list.ts:211]: this'll be added to the end of the PR chain

File: src/shared/orgListUtil.ts [hunk 4]
```diff
@@ -167,14 +180,22 @@ export class OrgListUtil {
...
+        currentValue = removeRestrictedInfoFromConfig(authInfo.getFields(true) as AuthFieldsFromFS);
       } catch (error) {
         // eslint-disable-next-line no-await-in-loop
         const logger = await OrgListUtil.retrieveLogger();
         logger.warn(`Error decrypting ${authInfo.getUsername()}`);
-        currentValue = OrgListUtil.removeRestrictedInfoFromConfig(authInfo.getFields());
+        currentValue = removeRestrictedInfoFromConfig(authInfo.getFields() as AuthFieldsFromFS);
+      }
+
+      if (!currentValue.username) {
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/shared/orgListUtil.ts:194]: Oh, I can take those out.  I fixed the type issues that made those necessary.

File: src/shared/orgListUtil.ts (hunk 1 above)
Comment (mshanemc) [reply src/shared/orgListUtil.ts:287]: The idea behind this retrieveLogger design: optimistically, we don't need to waste the (considerable!) time instantiating Loggers until we know we need one. 

Or maybe I'm missing the intent behind the suggestion

File: src/commands/force/org/list.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/commands/force/org/list.ts:211]: https://github.com/salesforcecli/plugin-org/pull/550/commits/cad2d86bdd98b1e21f06ea340a9627b7312ccf84

File: src/shared/orgListUtil.ts (hunk 2 above)
Comment (mshanemc) [reply src/shared/orgListUtil.ts:350]: https://github.com/salesforcecli/plugin-org/pull/550/commits/8989ccab2ba3ce247738b57a2afdf5fcf7783123

File: src/shared/orgListUtil.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/shared/orgListUtil.ts:194]: https://github.com/salesforcecli/plugin-org/pull/550/commits/8989ccab2ba3ce247738b57a2afdf5fcf7783123#diff-23272822541ec3f74596d4757354ae7de1ca9f60d6e4d90da55512bc4cdab76aL191

File: src/shared/orgListUtil.ts [hunk 5]
```diff
@@ -111,7 +134,7 @@ export class OrgListUtil {
       filename.match(/^00D.{15}\.json$/g)
     );
 
-    const allAuths: AuthInfo[] = await Promise.all(
+    const allAuths: Array<AuthInfo | undefined> = await Promise.all(
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/shared/orgListUtil.ts:137]: final assertion was on the return, not `allAuths` which still did have undefineds.  

While figuring this out, I made it better (I think).

**Review by peternhale — APPROVED** [review]

**Pull request merged**